src/libcrun/handlers/wamr.c

```diff
@@ -0,0 +1,249 @@
+  wasm_runtime_create_exec_env = dlsym (cookie, "wasm_runtime_create_exec_env");
+  wasm_runtime_call_wasm = dlsym (cookie, "wasm_runtime_call_wasm");
+  wasm_runtime_get_exception = dlsym (cookie, "wasm_runtime_get_exception");
+  wasm_runtime_destroy_exec_env = dlsym (cookie, "wasm_runtime_destroy_exec_env");
+  wasm_runtime_deinstantiate = dlsym (cookie, "wasm_runtime_deinstantiate");
+  wasm_runtime_unload = dlsym (cookie, "wasm_runtime_unload");
+  wasm_runtime_destroy = dlsym (cookie, "wasm_runtime_destroy");
+  wasm_runtime_get_wasi_exit_code = dlsym (cookie, "wasm_runtime_get_wasi_exit_code");
+  wasm_application_execute_main = dlsym (cookie, "wasm_application_execute_main");
+  wasm_runtime_set_wasi_args = dlsym (cookie, "wasm_runtime_set_wasi_args");
+
+  if (wasm_runtime_init == NULL)
+    error (EXIT_FAILURE, 0, "could not find wasm_runtime_init symbol in `libiwasm.so`");
+  if(wasm_runtime_full_init == NULL)
+    error (EXIT_FAILURE, 0, "could not find wasm_runtime_full_init symbol in `libiwasm.so`");
+  if (wasm_runtime_load == NULL)
+    error (EXIT_FAILURE, 0, "could not find wasm_runtime_load symbol in `libiwasm.so`");
+  if (wasm_runtime_instantiate == NULL) 
+    error (EXIT_FAILURE, 0, "could not find wasm_runtime_instantiate symbol in `libiwasm.so`");
+  if (wasm_runtime_lookup_function == NULL)
+    error (EXIT_FAILURE, 0, "could not find  wasm_runtime_lookup_functionsymbol in `libiwasm.so`");
+  if (wasm_runtime_create_exec_env == NULL)
+    error (EXIT_FAILURE, 0, "could not find  wasm_runtime_create_exec_env symbol in `libiwasm.so`");
+  if (wasm_runtime_call_wasm == NULL)
+    error (EXIT_FAILURE, 0, "could not find wasm_runtime_call_wasm symbol in `libiwasm.so`");
+  if (wasm_runtime_get_exception == NULL)
+    error (EXIT_FAILURE, 0, "could not find wasm_runtime_get_exception symbol in `libiwasm.so`");
+  if (wasm_runtime_destroy_exec_env == NULL)
+    error (EXIT_FAILURE, 0, "could not find wasm_runtime_destroy_exec_env symbol in `libiwasm.so`");
+  if (wasm_runtime_deinstantiate == NULL)
+    error (EXIT_FAILURE, 0, "could not find wasm_runtime_deinstantiate symbol in `libiwasm.so`");
+  if (wasm_runtime_unload == NULL)
+    error (EXIT_FAILURE, 0, "could not find wasm_runtime_unload symbol in `libiwasm.so`");
+  if (wasm_runtime_destroy == NULL)
+    error (EXIT_FAILURE, 0, "could not find wasm_runtime_destroy symbol in `libiwasm.so`");
+  if (wasm_runtime_get_wasi_exit_code == NULL)
+    error (EXIT_FAILURE, 0, "could not find wasm_runtime_get_wasi_exit_code symbol in `libiwasm.so`");
+  if (wasm_application_execute_main == NULL)
+    error (EXIT_FAILURE, 0, "could not find wasm_application_execute_main symbol in `libiwasm.so`");
+  if (wasm_runtime_set_wasi_args == NULL)
+    error (EXIT_FAILURE, 0, "could not find wasm_runtime_set_wasi_args symbol in `libiwasm.so`");
+
+  char *buffer, error_buf[128];
+  uint32_t size, stack_size = 8096, heap_size = 8096;
```

Comment:
Any specific reason for choosing `8096` as stack and heap size ?

Reply:
@flouthoc I followed this [example](https://github.com/bytecodealliance/wasm-micro-runtime/blob/main/doc/embed_wamr.md)